src/main/java/service/impl/ReportWriterImpl.java

```diff
@@ -0,0 +1,21 @@
+package service.impl;
+
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import service.ReportWriter;
+
+public class ReportWriterImpl implements ReportWriter {
+
+    @Override
+    public void writeReport(String report, String toFileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            writer.write(report);
+        } catch (FileNotFoundException e) {
```

Comment:
You don't need to catch FileNotFoundException separately because FileWriter constructor which can throw this exception is already inside try-with-resources statement. So you can remove this catch block.